### PR TITLE
feat: custom derivation paths in Mnemonic ECDSA private key derivation

### PR DESCRIPTION
**Pull request by bguiz**
**Description**:

## What

- Add custom derivation path capability

## Why

- Compatibility/ parity with EVM dev tools + wallets. Specifically:
  - When working Metamask, you need to use the hardcoded derivation path in it, and in SDK there's also a (different) hardcoded derivation path.
  - In EVM dev tools - web3.js/ethers.js/viem etc - they allow you to specify a custom derivation path, and do not force you to use the Ethereum one.
- The lack of support for custom derivation paths is a friction point for EVM developers onboarding into Hedera development
  - This may have the effect of effectively discouraging use of the SDK, in favour of using 3rd party libraries that do support custom derivation paths
- we should aim to reduce the reliance on 3rd party dependencies in general, within code that is written using the SDK
  - The following examples demonstrate that (1) additional code is needed, and (2) additional 3rd party dependencies need to be brought in:
  - example of extra code required in JS: https://github.com/hedera-dev/scratch-hello-world/blob/60fa4007/js-hello-world/00-create-fund-account/script-create-fund-account.js#L19-L23
  - example of extra code required in Java: https://github.com/hedera-dev/scratch-hello-world/blob/60fa4007/java-hello-world/00-create-fund-account/src/main/java/createfundaccount/ScriptCreateFundAccount.java#L30-L55
  - example of extra code required in Go: https://github.com/hedera-dev/scratch-hello-world/blob/60fa4007/golang-hello-world/00-create-fund-account/script-create-fund-account.go#L37-L61

**Related issue(s)**:

Fixes: https://github.com/hashgraph/hedera-sdk-java/issues/1263#issuecomment-1330953933

**Notes for reviewer**:

- Related: https://github.com/hedera-dev/scratch-hello-world/pull/1
- Discussion with @thenswan + @0xivanov: https://swirldslabs.slack.com/archives/C0453UVBR42/p1717898573711519?thread_ts=1717061977.081859&cid=C0453UVBR42
- Related: https://github.com/hashgraph/hedera-sdk-java/pull/1842
- Related: https://github.com/hashgraph/hedera-sdk-go/pull/958

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: src/Mnemonic.js [hunk 1]
```diff
@@ -136,6 +138,57 @@ export default class Mnemonic {
...
+    /**
+     * Common implementation for both `toStandardECDSAsecp256k1PrivateKey`
+     * functions.
+     *
+     * @param {string} passphrase  the passphrase used to protect the
+     *                             mnemonic, use "" for none
+     * @param {Array<number>} derivationPathValues derivation path as an
+     *                             integer array,
+     *                             see: `calculateDerivationPathValues`
+     * @returns a private key
```
Comment (svetoslav-nikol0v) [inline src/Mnemonic.js:183]:
```suggestion
     * @returns {Promise<PrivateKey>}
```

**Review by svetoslav-nikol0v — CHANGES_REQUESTED** [review]

File: src/Mnemonic.js (hunk 1 above)
Comment (bguiz) [reply src/Mnemonic.js:183]: Thanks for the suggestion!

File: src/Mnemonic.js [hunk 2]
```diff
@@ -136,6 +138,57 @@ export default class Mnemonic {
...
+     */
+    async toStandardECDSAsecp256k1PrivateKeyImpl(
+        passphrase,
+        derivationPathValues,
+    ) {
+        return this.toEcdsaPrivateKey(passphrase, derivationPathValues);
```
Comment (svetoslav-nikol0v) [inline src/Mnemonic.js:189]: Here, `await` should be added as well

Comment (svetoslav-nikol0v) [inline src/Mnemonic.js:189]:
```suggestion
        return await this.toEcdsaPrivateKey(passphrase, derivationPathValues);
```

**Review by svetoslav-nikol0v — CHANGES_REQUESTED** [review]

File: src/Mnemonic.js (hunk 2 above)
Comment (bguiz) [reply src/Mnemonic.js:189]: Thanks!

Comment (bguiz) [reply src/Mnemonic.js]: @svetoslav-nikol0v Thanks - I have implemented both of the changes requested in [1708ea5](https://github.com/hashgraph/hedera-sdk-js/pull/2341/commits/1708ea56b9e6212529b698807c8ba9ae9f638e90)

**Comment by bguiz** [comment]
@svetoslav-nikol0v + @thenswan [5a5b32d2](https://github.com/hashgraph/hedera-sdk-js/pull/2341/commits/5a5b32d2fd6af054c78845c6fd5853f4ed05284a)

Attempt to increase code coverage of `src/Mnemonic.js` --> https://app.codecov.io/gh/hashgraph/hedera-sdk-js/pull/2341?src=pr&el=tree#23d3b499f82e11bd737b0faf0152911f-R153

Further context: https://swirldslabs.slack.com/archives/C0453UVBR42/p1719635439040999?thread_ts=1717061977.081859&cid=C0453UVBR42

**Review by svetoslav-nikol0v — APPROVED** [review]

**Pull request merged**